app.js

```diff
@@ -20,13 +21,17 @@ app.get("/api/articles", getArticles);
 
 app.get("/api/articles/:article_id/comments", getCommentsByArticle);
 
+app.post("/api/articles/:article_id/comments", postComment);
+
 app.all("*", (req, res) => {
   res.status(404).send({ msg: "path not found" });
 });
 
 app.use((err, req, res, next) => {
   if (err.code === "22P02" || err.code === "23502") {
     res.status(400).send({ msg: "bad request" });
+  } else if (err.code === "23503") {
+    res.status(404).send({ msg: "not found" });
```

Comment:
It might be time to consider separating this out to error handlers as a long if else chain may get hard to track errors. For a steer on how to do this checkout the bottom of the error handler notes.